File: services/python_services/dbConnect.py
import os
import asyncio
import asyncpg as pg
from dotenv import load_dotenv

env_path = '../attend_wise/shared/utils/.env'
load_dotenv(dotenv_path=env_path)

DB_PORT = os.getenv("LOCAL_POSTGRESS_PORT")
DB_NAME = os.getenv("LOCAL_POSTGRESS_DATABASE")
DB_USER = os.getenv("LOCAL_POSTGRESS_USER")
DB_PASS = os.getenv("LOCAL_POSTGRESS_PASSWORD")

# print(f'Connecting to {DB_NAME} at {DB_PORT} as {DB_USER}')

isConnected = False
db = None

async def dbConnect():
    global isConnected, db
    if isConnected and db is not None:
        print("Already connected to the database.")
        return
    
    try:
        db = await pg.connect(
            user=DB_USER,
            password=DB_PASS,
            database=DB_NAME,
            host="localhost",
            port=DB_PORT,
        )
        
        print("Connected to the database successfully.")
        isConnected = True
        return db
    except pg.PostgresError as e:
        print(f"Error connecting to the database: {e}")
        isConnected = False
        return None
asyncio.run(dbConnect())

# def main():
#     global DB_USER
#     print(DB_USER)
#     async def test_connection():
#         await dbConnect()

#     # asyncio.run(test_connection())

# if __name__ == "__main__":
#     main()

File: services/js_services/dbConnect.js
import dotenv from 'dotenv';
import pgPromise from 'pg-promise';

dotenv.config({ path: 'D:/CODING/Devlopment/PROJECTS/attend_wise/shared/utils/.env' });
const pgp= pgPromise();

let db = null;
let isConnected = false;

async function dbConnect(){
    if (isConnected && db){
        console.log("DataBase is already connected")
        return db;
    }

    try{
        db = pgp(`postgres://${process.env.LOCAL_POSTGRESS_USER}:${process.env.LOCAL_POSTGRESS_PASSWORD}@localhost:${process.env.LOCAL_POSTGRESS_PORT}/${process.env.LOCAL_POSTGRESS_DATABASE}`);
        
        // checks if db connection with light weight querry
        await db.one('SELECT 1')
            .then(() => {
                isConnected = true;
                console.log("DataBase connected sucessfully")
            });
    }catch(error){
        console.log("!Error connecting to the database:: ", error)
    }        
}

export default dbConnect;